src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,201 @@
+        }
+    }
+
+    /**
+     * Parses <code>Task</code> objects into strings to be written to the
+     * file on the hard disk.
+     *
+     * @param task <code>Task</code> object to be parsed
+     * @return String representation of the task. Note that this string is
+     *          different from the string returned by <code>task.toString()</code>.
+     *          e.g. "[T][✓] Homework" from <code>task.toString()</code> will be
+     *          represented as "T | 1 | Homework".
+     * @throws DukeException If the type of the task cannot be recognised.
+     */
+    protected String parseToStorage(Task task) throws DukeException {
+        String taskType = "";
+        String status = task.isDone() ? "1" : "0"; // 1 for done and 0 for not done
+        String taskDescription = "";
+
+        if (task instanceof Todo) {
+            taskType = "T";
+            taskDescription = task.getTaskName();
+        } else if (task instanceof Deadline) {
+            taskType = "D"; // + date
+            taskDescription = task.getTaskName() +
+                " | " +
+                ((Deadline) task).getDateTime().format(
+                    DateTimeFormatter.ofPattern("d MMM yyyy, h.m a"));
+        } else if (task instanceof Event) {
+            taskType = "E";
+            taskDescription = task.getTaskName() +
+                " | " +
+                ((Event) task).getDateTime().format(
+                    DateTimeFormatter.ofPattern("d MMM yyyy, h.m a"));
+        } else {
+            throw new DukeException("Cannot recognise type");
+        }
+
+        return taskType +
+            " | " +
+            status +
+            " | " +
+            taskDescription;
+    }
+
+    /**
+     * Parses string representation of task stored in the file on the hard disk
+     * to create a <code>Task</code> object.
+     *
+     * @param storedTask String representation of the task
+     * @return Task represented by the string input.
+     * @throws DukeException If the string is in the wrong format.
+     */
+    protected Task parseFromStorage(String storedTask) throws DukeException {
+        String[] taskElements = storedTask.split(" \\| ", 4);
+
+        try {
+            Task task = null;
+            if (storedTask.charAt(0) == 'T') {
+                task = new Todo(taskElements[2]);
+            } else if (storedTask.charAt(0) == 'D') {
+                String taskName = taskElements[2];
+                LocalDateTime dateTime = LocalDateTime.parse(taskElements[3],
+                    DateTimeFormatter.ofPattern("d MMM yyyy, h.m a"));
+                task = new Deadline(taskName, dateTime);
+            } else if (storedTask.charAt(0) == 'E') {
+                String taskName = taskElements[2];
+                LocalDateTime dateTime = LocalDateTime.parse(taskElements[3],
+                    DateTimeFormatter.ofPattern("d MMM yyyy, h.m a"));
+                task = new Event(taskName, dateTime);
```

Comment:
Perhaps this (and other similar chunks of code) should be broken up into distinct logic units with comments to describe what they do?